java/compaction/compaction-core/src/test/java/sleeper/compaction/strategy/CompactionStrategyIndexTest.java

```diff
@@ -0,0 +1,135 @@
+            FileReferenceFactory factory = FileReferenceFactory.from(partitionsBuilder.buildTree());
+            FileReference file1 = factory.rootFile("file1.parquet", 456L);
+            FileReference file2 = factory.rootFile("file2.parquet", 789L);
+            FileReference file3 = factory.rootFile("file3.parquet", 123L);
+            List<FileReference> allFileReferences = List.of(file1, file2, file3);
+
+            // When
+            CompactionStrategyIndex index = new CompactionStrategyIndex(tableStatus, allFileReferences, partitionsBuilder.buildList());
+
+            // Then
+            assertThat(index.getFilesInLeafPartitions())
+                    .containsExactly(unassignedFilesInPartition("root", List.of(file3, file1, file2)));
+        }
+
+        @Test
+        void shouldIndexMultipleFilesWithSameNumberOfRecords() {
+            // Given
+            PartitionsBuilder partitionsBuilder = new PartitionsBuilder(schema)
+                    .rootFirst("root");
+            FileReferenceFactory factory = FileReferenceFactory.from(partitionsBuilder.buildTree());
+            FileReference file1 = factory.rootFile("file1.parquet", 100L);
+            FileReference file2 = factory.rootFile("file2.parquet", 100L);
+            FileReference file3 = factory.rootFile("file3.parquet", 100L);
+            List<FileReference> allFileReferences = List.of(file1, file2, file3);
+
+            // When
+            CompactionStrategyIndex index = new CompactionStrategyIndex(tableStatus, allFileReferences, partitionsBuilder.buildList());
+
+            // Then
+            assertThat(index.getFilesInLeafPartitions())
+                    .containsExactly(unassignedFilesInPartition("root", List.of(file1, file2, file3)));
+        }
+
+        @Test
+        void shouldIndexMultipleLeafPartitionsWithMultipleFiles() {
+            // Given
+            PartitionsBuilder partitionsBuilder = new PartitionsBuilder(schema)
+                    .rootFirst("root")
+                    .splitToNewChildren("root", "L", "R", 123L);
+            FileReferenceFactory factory = FileReferenceFactory.from(partitionsBuilder.buildTree());
+            FileReference file1 = factory.partitionFile("L", "file1.parquet", 120L);
+            FileReference file2 = factory.partitionFile("R", "file2.parquet", 456L);
+            FileReference file3 = factory.partitionFile("R", "file3.parquet", 789L);
+            List<FileReference> allFileReferences = List.of(file1, file2, file3);
+
+            // When
+            CompactionStrategyIndex index = new CompactionStrategyIndex(tableStatus, allFileReferences, partitionsBuilder.buildList());
+
+            // Then
+            assertThat(index.getFilesInLeafPartitions())
+                    .containsExactlyInAnyOrder(
+                            unassignedFilesInPartition("L", List.of(file1)),
+                            unassignedFilesInPartition("R", List.of(file2, file3)));
+        }
+    }
```

Comment:
There were tests for assigned files as well here. I'm not sure how important it is, since it's still tested indirectly, but did you want to bring that back?